Add support for HTTPS servers using Zephyr

taltenbach opened the pull request:
This PR proposes modifications to add the support for running HTTPS servers with Zephyr using Zephyr Secure Sockets.

Using Secure Sockets enables in particular to easily supply the TLS certificate and private key since it is enough to add them to TLS Credentials subsystem and to provide the corresponding tag through the CivetWeb configuration.

Note that TLS support is not added for the HTTP client API.

taltenbach committed:
Add HTTPS server support with Zephyr

Add the support for running HTTPS servers with Zephyr using Zephyr
Secure Sockets. Using Secure Sockets enables in particular to easily
supply the TLS certificate and private key since it is enough to add
them to TLS Credentials subsystem and to provide the corresponding tag
through the CivetWeb configuration.

Note that TLS support is not added for the HTTP client API.

Signed-off-by: Thomas Altenbach <taltenbach@witekio.com>

nvincent-vossloh commented:
> Note that TLS support is not added for the HTTP client API.

Why ? Is it fundamentally different to support TLS through the client API ?

edit: when you mentionned "HTTP client API" did you meant, the civetweb server which can perform HTTP client API request, or the API that you developped in SiemaApplications/vog-zephyr-nodes#140 ?

taltenbach committed:
Ensure SSL certificate tag not NULL

In case the user did not provided a SSL certificate when running a
server using TLS with Zephyr, atoi will be with NULL. Avoid that issue
by checking first whether the SSL certificate tag is NULL.

taltenbach commented:
I meant the HTTP client API provided by CivetWeb. I didn't take time to implement support for TLS through that API since you do not use it and Zephyr seems to support performing HTTPS requests to a server without using CivetWeb

nvincent-vossloh reviewed (approved):
Ok for me. I'll let you merge unless you have additional commits.
